spec/query_spec.rb

```diff
@@ -62,29 +40,122 @@
+      [
+        seen_but_not_cached_query,
+        Readyset::Query.new({
+          'query id' => 'q_8892818e62c34ecd',
+          'proxied query' => 'SELECT * FROM "t" WHERE ("y" = $1)',
+          'readyset supported' => 'yes',
+          'count' => 5,
+        }),
+      ]
+    end
+    let(:unsupported_or_pending_queries) do
+      [
+        Readyset::Query.new(
+          'query id' => 'q_f9bfc11a043b2f75',
+          'proxied query' => 'SHOW TIME ZONE',
+          'readyset supported' => 'unsupported',
+          'count' => 5,
+        ),
+        Readyset::Query.new(
+          'query id' => 'q_d7cbfb8a03d589cf',
+          'proxied query' => 'SELECT * FROM "t" WHERE ("x" < 1)',
+          'readyset supported' => 'pending',
+          'count' => 5,
+        ),
+      ]
+    end
+
+    before do
+      allow(Readyset::Query).to receive(:all_seen_but_not_cached).and_return(queries)
+    end
+
+    context 'when every Readyset::Query#cache! invocation succeeds' do
+      before do
+        queries.each { |query| allow(query).to receive(:cache!).with(always: true) }
+        subject
+      end
+
+      it 'invokes Readyset::Query#cache! on every supported query with the given "always" ' \
+        'parameter' do
+        supported_queries.each do |query|
+          expect(query).to have_received(:cache!).with(always: true)
+        end
+      end
+
+      it 'does not invoke Readyset::Query#cache! on any unsupported or pending queries' do
+        unsupported_or_pending_queries.each do |query|
+          expect(query).not_to have_received(:cache!)
+        end
+      end
+
+      it 'returns nil' do
+        is_expected.to be_nil
+      end
+    end
+
+    context 'when one of the Readyset::Query#cache! invocations fails' do
+      before do
+        allow(queries[0]).to receive(:cache!).and_raise(StandardError)
+        allow(queries[1]).to receive(:cache!)
+
+        begin
+          subject
+        rescue StandardError
+          nil
+        end
+      end
+
+      it 'raises the error raised by the Readyset::Query#cache! invocation' do
+        expect { subject }.to raise_error(StandardError)
+      end
+
+      it 'invokes Readyset::Query#cache! on the queries in the list up to and including the ' \
+        'query that caused the error with the given "always" parameter' do
+        expect(queries[0]).to have_received(:cache!).with(always: true)
+      end
+
+      it 'does not invoke Readyset::Query#cache! on any of the queries in the list after the ' \
+        'query that caused the error' do
+        expect(queries[1]).not_to have_received(:cache!)
+      end
+    end
+  end
+
+  describe '.drop_all_caches!' do
+    subject { Readyset::Query.drop_all_caches! }
+
+    before do
+      allow(Readyset).to receive(:raw_query).with('DROP ALL CACHES')
+
+      subject
+    end
+
+    it 'invokes "DROP ALL CACHES" against Readyset' do
+      expect(Readyset).to have_received(:raw_query).with('DROP ALL CACHES')
```

Comment:
So, in regards to *what's* being tested, would it be accurate to say this:
- ensure that the correct SQL command is being executed.
  - *structure*: `VERB  modifier NOUN`
  - *arguments*: `query_id`(s)
- sending the `query_id` to Readyset as a parameter. 

So, at least for the class methods, the SQL command sent is defined and we're offering an easy way of invoking said command. 

While for the instance methods, such as `#cache!`, it's also surfacing information to the user, such as potential errors, or that `#reload` ran after receiving a server response.

---
another relevant thoughtbot article:  [Testing Techniques: The I/O Table](https://thoughtbot.com/blog/testing-techniques-the-i-o-table)

Reply:
Yes, I think that's an accurate characterization!